### DISCUSSION TITLE
Thoughts on an aggregated media guide

### DISCUSSION THREAD

**Discussion by Spartan51150** [PRProd/HA-Firemote#418] [General]
Essentially I would love to have a media guide browsing functionality where you can see what is on your streaming services

Doesn't anyone else think this is even remotely achievable and if so could it integrate with this firemote to launch shows or at least the streaming app when you find something you want to watch.

**Comment by PRProd** [comment]
@Spartan51150 ~

That's an interesting idea for a VERY ambitious project!  A media guide like that is well beyond my personal abilities right now, so I cannot answer your question about that other than saying, "not me!"
<br>

As far as this part goes
> if so could it integrate with this firemote to launch shows or at least the streaming app when you find something you want to watch.

If you happened across one or more of these things, they will give you a really big hint:
 - [How Does Firemote Send commands to my devices?](https://github.com/PRProd/HA-Firemote/wiki#how-does-firemote-send-commands-to-my-devices)
 - [Expose Buttons to Remote Send Command Services #102](https://github.com/PRProd/HA-Firemote/discussions/102)
 - [Using ADB Commands in Automations to Emulate FIremote #338](https://github.com/PRProd/HA-Firemote/discussions/338)
 - [Android App development support](https://github.com/PRProd/HA-Firemote/discussions/376#discussioncomment-9157189)
 
 Essentially, Firemote isn't an integration.  Firemote is just a card.  Because it is a card, it relies on Home Assistant integrations such as ADB, Android TV Remote, Apple TV, Roku, etc.  What Firemote does is simply send Home Assistant commands to these existing integrations to achieve the desired result.  There isn't actually a way for any outside script, integration, or card to "hook into" Firemote directly.  If someone was to take on the enormous task of making a media guide, they could look at the Firemote codebase to learn how to send commands, but they couldn't *use* Firemote to send the commands for them.

I hope that made sense.

-Doug
